scripts/check_site.lua

```diff
@@ -108,14 +107,27 @@ local function var_error(path, val, msg)
 		src = site_src()
 	end
 
+	return src
+end
+
+local function var_error(path, val, msg)
+	if type(val) == 'string' then
+		val = string.format('%q', val)
+	end
+
 	local found = 'unset'
 	if val ~= nil then
 		found = string.format('%s (a %s value)', tostring(val), type(val))
 	end
 
-	config_error(src, 'expected %s to %s, but it is %s', path_to_string(path), msg, found)
+	config_error(conf_src(path), 'expected %s to %s, but it is %s', path_to_string(path), msg, found)
 end
 
+function obsolete(path)
```

Comment:
allow for custom obsoletion messages to be passed

Reply:
added the possibility to add custom messages